docs/how-to/tuning-guides/mi300x.rst

```diff
@@ -0,0 +1,1425 @@
+   each with 32 KB L1 cache, a Unified Compute System with 4 ACE Compute
+   Accelerators, shared 4MB of L2 cache, and an HWS Hardware Scheduler.
+
+You can query hardware resources with the command ``rocminfo`` in the
+``/opt/rocm/bin`` directory. For instance, query the number of CUs, number of
+SIMD, and wavefront size using the following commands.
+
+.. code-block:: shell
+
+   rocminfo | grep "Compute Unit"
+
+   rocminfo | grep "SIMD"
+
+   rocminfo | grep "Wavefront Size"
+
+For the MI300X, the goal is to have a minimum of 1024 thread
+blocks or workgroups in the grid (kernel), with a preference for
+more.
+
+Identifying additional parallelism within the algorithm is necessary to
+enhance GPU utilization. For more information and examples, see
+`Accelerating A Triton Fused Kernel For W4a16 Quantized Inference With
+SplitK Work Decomposition <https://arxiv.org/pdf/2402.00025v1>`__.
+
+Auto-tunable kernel configurations and environment variables
+------------------------------------------------------------
+
+Auto-tunable kernel configuration is the memory access and computation
+assigned to each compute unit. It is related to using
+:ref:`LDS <mi300x-cu-fig>`, register, and scheduling different tasks on a CU.
+
+The accelerator or GPU contains global memory, local data share (LDS), and
+registers. Global memory has high access latency, but is large. LDS access has
+much lower latency, but is smaller; it is a fast on-CU software-managed memory
+that can be used to efficiently share data between all work items in a block.
+Register access is the fastest yet smallest among the three.
+
+.. _mi300x-cu-fig:
+
+.. figure:: ../../data/shared/compute-unit.png
+
+   Schematic representation of a CU in the CDNA2 or CDNA3 architecture.
+
+The following is a list of kernel arguments used for tuning.
+
+``num_stages=n``
+   Adjusts the number of pipeline stages for different types of kernels. On AMD accelerators, set ``num_stages``
+   according to the following rules:
+
+   * For kernels with a single GEMM, set to ``0``.
+
+   * For kernels with two GEMMs fused (Flash Attention, or any other kernel
+     that fuses 2 GEMMs), set to ``1``.
+
+   * For kernels that fuse a single GEMM with another non-GEMM operator
+     (for example ReLU activation), set to ``0``.
+
+   * For kernels that have no GEMMs, set to ``1``.
+
+``waves_per_eu=n``
+   Helps to manage Vector General Purpose Registers (VGPR) usage to achieve
+   desired occupancy levels. This argument hints to the compiler to reduce VGPR
+   to achieve ``n`` occupancy where ``n`` is a number. The goal is to achieve a
+   certain occupancy level for each Execution Unit (EU, also called
+   :ref:`SIMD Unit <mi300x-cu-fig>`) to achieve better latency or throughput.
+   See :ref:`mi300x-compute-kernel-occ` for more information about how to
+   compute occupancy. 
+
+   This argument is useful if:
+
+   * The occupancy of the kernel is limited by VGPR usage.
```

Comment:
```suggestion
   * The occupancy of the kernel is limited by VGPR usage, and
```
Since the tune has changed from original documentation, please make sure we have the "and" here so that it means both conditions need to be met.

Reply:
Done